src/decoder_libsndfile.cpp

```diff
@@ -139,9 +139,19 @@ int LibsndfileDecoder::FillBuffer(uint8_t* buffer, int length) {
 			break;
 		case Format::S32:
 			{
+				// This code assumes that int32_t is int but on 3ds it is long int
+				// Both types are 4 byte in size but of incompatible pointer types
+#ifdef _3DS
+				decoded=sf_read_int(soundfile,(int*)buffer,length/sizeof(int));
+#else
 				decoded=sf_read_int(soundfile,(int32_t*)buffer,length/sizeof(int32_t));
-				if(!decoded) finished=true;
+#endif
+                if(!decoded) finished=true;
+#ifdef _3DS
+				decoded*=sizeof(int);
+#else
 				decoded*=sizeof(int32_t);
+#endif
 			}
```

Comment:
@Ghabry 
Yeah it does, but why not drop int32_t altogether (no ifdef) and use int instead for all architectures.
The interface of libsndfile uses int anyway. [sf_count_t sf_read_int (SNDFILE *sndfile, int *ptr, sf_count_t items); ]

I just used int32_t for clarification (and to stay uniform with the int16_t above.)